pkg/karmadactl/cmdinit/karmada/check.go

```diff
@@ -14,19 +15,39 @@ import (
 
 // WaitAPIServiceReady wait the api service condition true
 func WaitAPIServiceReady(c *aggregator.Clientset, name string, timeout time.Duration) error {
-	if err := wait.PollImmediate(time.Second, timeout, func() (done bool, err error) {
-		apiService, e := c.ApiregistrationV1().APIServices().Get(context.TODO(), name, metav1.GetOptions{})
-		if e != nil {
-			return false, nil
+	apiService, err := c.ApiregistrationV1().APIServices().Get(context.TODO(), name, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	opts := metav1.SingleObject(apiService.ObjectMeta)
+	t := int64(timeout.Seconds())
+	opts.TimeoutSeconds = &t
+	apiServiceWatcher, err := c.ApiregistrationV1().APIServices().Watch(context.Background(), opts)
+	if err != nil {
+		return err
+	}
+
+	for {
+		event, ok := <-apiServiceWatcher.ResultChan()
+		if !ok {
+			return fmt.Errorf("Waiting for APIService(%s) condition(%s) timeout", name, apiregistrationv1.Available)
 		}
-		if apiregistrationv1helper.IsAPIServiceConditionTrue(apiService, apiregistrationv1.Available) {
-			return true, nil
+
+		u, err := runtime.DefaultUnstructuredConverter.ToUnstructured(event.Object)
+		if err != nil {
+			return err
+		}
+
+		updatedAPIService := &apiregistrationv1.APIService{}
+		if err := runtime.DefaultUnstructuredConverter.FromUnstructured(u, updatedAPIService); err != nil {
+			return err
+		}
+
+		if apiregistrationv1helper.IsAPIServiceConditionTrue(updatedAPIService, apiregistrationv1.Available) {
+			return nil
 		}
 
 		klog.Infof("Waiting for APIService(%s) condition(%s), will try", name, apiregistrationv1.Available)
```

Comment:
Here will dump `info` logs, does it seems too verbose?

Reply:
The length looks ok, for example `I0601 05:59:14.497568 20024 check.go:26] Waiting for APIService(v1alpha1.cluster.karmada.io) condition(Available), will try`

Maybe we shouldn't print this log, it just tells the user that karmadactl is executing and the current shell is not disconnected, so it's not really that useful

PS: I prefer to use kwokctl circle waiting method, but this seems to be a bit of a big change, I don't know what you think about the evolution of karmadactl later

Reply:
The `kworkctl circle` is cool, many project CLI use this way, like [subctl of submariner](https://submariner.io/getting-started/quickstart/kind/#install-subctl) that are interesting and clear. I think we also use this way in the future.

![image](https://kwok.sigs.k8s.io/img/demo/audit-log.svg)